lib/newFileMenu.ts

```diff
@@ -0,0 +1,80 @@
+import logger from "./utils/logger"
+
+export enum FileType {
+	FILE = 'file',
+	FOLDER = 'folder',
+}
+
+export interface Entry {
+	id: string
+	displayName: string
+	templateName: string
+	iconSvgInline?: string
+	iconClass?: string
+	fileType: FileType
+	handler: Function
+}
+
+export class NewFileMenu {
+	private _entries: Array<Entry> = []
+	
+	public registerEntry(entry: Entry) {
+		this.validateEntry(entry)
+		this._entries.push(entry)
+	}
+
+	public unregisterEntry(entry: Entry | string) {
+		const entryIndex = typeof entry === 'string'
+			? this.getEntryIndex(entry)
+			: this.getEntryIndex(entry.id)
+		
+		if (entryIndex === -1) {
+			logger.warn('Entry not found, nothing removed', { entry , entries: this.getEntries() })
+		}
+
+		this._entries.splice(entryIndex, 1)
```

Comment:
Will boom when `entryIndex === -1`: `this._entries.splice(-1, 1)`.

Best would be to return in the `if`.

Reply:
> Will boom when `entryIndex === -1`: `this._entries.splice(-1, 1)`.
> 
> Best would be to return in the `if`.

Already caught three lines above?
![image](https://user-images.githubusercontent.com/14975046/173625957-a649479b-30af-4333-aa9e-ce3167d4b8b8.png)

Reply:
Ah, I'm missing a return, sorry, I misread

Reply:
Done!